File: src/main/java/com/application/cv_application/requests/ApplicationRequest.java
package com.application.cv_application.requests;

import java.time.LocalDate;

public record ApplicationRequest(
        LocalDate appliedDate,
        LocalDate updatedDate,
        Integer candidateId,
        Integer jopId,
        String status,
        String firstname,
        String lastname,
        String phoneNumber,
        String address,
        String nationality,
        String gender,
        LocalDate dateOfBirth
) {}

File: src/app/service/application-request.ts
export interface ApplicationRequest {
  applicationName: string;
  appliedDate: string; // or Date
  updatedDate: string; // or Date
  status: string;
  candidateId: number; // Changed from number | null to number (required)
  jopId: number; // Changed from number | null to number (required)
}
